### ISSUE TITLE
Flutter Web: Radio buttons require proper keyboard interactions (Accessibility)

### ISSUE THREAD

**Issue by jcollins-disney** [flutter/flutter#162137]
### Use case

Radio buttons (contained in a radiogroup) do not have the expected keyboard interactions.

### Proposal

Add the following keyboard interactions for radio buttons contained inside of a radiogroup:

- **Tab** and **Shift + Tab**: Move focus into and out of the radio group. When focus moves into a radio group:
-- If a radio button is checked, focus is set on the checked button.
-- If none of the radio buttons are checked, focus is set on the first radio button in the group.
- **Space**: checks the focused radio button if it is not already checked.
- **Right Arrow** and **Down Arrow**: move focus to the next radio button in the group, uncheck the previously focused button, and check the newly focused button. If focus is on the last button, focus moves to the first button.
- **Left Arrow** and **Up Arrow**: move focus to the previous radio button in the group, uncheck the previously focused button, and check the newly focused button. If focus is on the first button, focus moves to the last button.

**Comment by chunhtai** [comment]
This may require rework of radio and radio group. Currently framework doesn't have a notion of radio group

**Comment by chunhtai** [comment]
duplicates of https://github.com/flutter/flutter/issues/113562
